### ISSUE TITLE
columnar table without compression seems to be larger than the origin one

### ISSUE THREAD

**Issue by DeoLeung** [citusdata/citus#5997]
Hi, we are testing our database using columnar tables, if we try using columnar without compression, the result table size seems to be bigger than the origin one

```sql
-- test compression
-- create columnar table
create TABLE collector."a-20220527" USING columnar as 
SELECT * FROM collector."a";
VACUUM VERBOSE collector."a-20220527";
-- output
INFO:  statistics for "a-20220527":
storage id: 10000000000
total file size: 1124384768, total data size: 1120975712
compression rate: 7.99x
total row count: 2857002, stripe count: 20, average rows per stripe: 142850
chunk count: 3146, containing data for dropped columns: 0, zstd compressed: 3146
VACUUM
-- non compress columnar table
create TABLE collector."a-20220531" USING columnar
as TABLE collector."a" WITH no data;
SELECT alter_columnar_table_set(
  'collector.a-20220531',
  compression => 'none');
INSERT INTO collector."a-20220531"
SELECT * FROM collector."a";
VACUUM VERBOSE collector."a-20220531";
-- output
INFO:  statistics for "a-20220531":
storage id: 10000000001
total file size: 8982011904, total data size: 8955590598
compression rate: 1.00x
total row count: 2857002, stripe count: 20, average rows per stripe: 142850
chunk count: 2318, containing data for dropped columns: 0, none compressed: 2318
VACUUM
-- compare size
SELECT pg_size_pretty( pg_total_relation_size('collector.a') ) AS origin, pg_size_pretty(pg_total_relation_size('collector.a-20220527')) AS citus,
pg_size_pretty(pg_total_relation_size('collector.a-20220531')) AS non_compress;

--- output
| origin |citus |non\_compress |
| --- | --- | --- |
| 6087 MB |1072 MB |8566 MB |

-- a defnition
CREATE TABLE collector."a" (
    id integer NOT NULL,
    action_time timestamp with time zone NOT NULL DEFAULT now(),
    create_time timestamp with time zone NOT NULL DEFAULT now(),
    update_time timestamp with time zone NOT NULL DEFAULT now(),
    account_id integer,
    ip cidr NOT NULL,
    lng double precision,
    lat double precision,
    extra jsonb NOT NULL DEFAULT '{}'::jsonb,
    operator integer ,
    action text
);

-- Indices -------------------------------------------------------

CREATE INDEX "ix_a-_f020" ON collector."a"(id int4_ops);
CREATE INDEX "ix_a-_db19" ON collector."a"(operator int4_ops);
CREATE INDEX "ix_a_account_id" ON collector."a"(account_id int4_ops);
```

thanks
----------------
using postgres 13.7, citus 10.2.4

**Comment by thanodnl** [comment]
Initial thoughts here are that the jsonb data you are storing hits toast tables in postgres. This field would then compress when stored in heap tables within postgres.

Any reason to use columnar without compression? One of the design goals, if not the most important one, is to be able to compress tables. Using `none` compression takes that benefit away :).

**Comment by DeoLeung** [comment]
yes we are using it for compression of our history table and it works well

the `none` just for testing reason, we want to made the origin table columnar, to test if without compression, we could gain some speed without costing extra cpu(for decompression I think). however since it's not working with our `pgroonga` index on jsonb, we put it off at the moment :)

**Comment by thanodnl** [comment]
Thanks for the insights.

pgroonga is an interesting index, might be worth to track together with some other indexes we have in mind for columnar. Unfortunately indexes on columnar aren't super efficient. After a very small amount of hits in the index we have noticed it becomes cheaper to run a full table scan on columnar compared to using the index for access. Due to a combination of correlation between subsequent index entries and the fact a complete chunk needs decompressing for a single index hit.

With regards to the size, I don't think the size is a bug, or anything actionable for us.

Closing this issue.
